### PR TITLE
Fix flashloan inflation trick

### LINKED ISSUE #848: Sync repo

### PR DESCRIPTION
**Pull request by Rubilmax**
This branch contains changes applied through the latest upgrade of Morpho-AaveV3 contracts

Fixes #848

### CODE REVIEW

**Review by MerlinEgalite — DISMISSED** [review]

**Review by Tristan22400 — DISMISSED** [review]

**Review by pakim249CAL — DISMISSED** [review]

File: test/integration/TestIntegrationWithdrawCollateral.sol
Comment (MathisGD) [inline test/integration/TestIntegrationWithdrawCollateral.sol:1]: A test that the p2pIndexes are correctly updated could be added (similar to what I have done in ma2 and mc).

**Review by MerlinEgalite — DISMISSED** [review]

**Review by MathisGD — DISMISSED** [review]

File: test/integration/TestIntegrationMorphoGetters.sol [hunk 1]
```diff
@@ -92,4 +94,32 @@ contract TestIntegrationMorphoGetters is IntegrationTest {
             "poolBorrowIndex != truePoolBorrowIndex"
         );
     }
+
+    function testPoolSupplyIndexGrowthInsideBlock(uint256 seed) public {
+        TestMarket storage market = testMarkets[_randomCollateral(seed)];
```
Comment (Tristan22400) [inline test/integration/TestIntegrationMorphoGetters.sol:99]: Why are we taking only collateral asset? I suggest to use all suppliable assets?

File: test/integration/TestIntegrationMorphoGetters.sol [hunk 2]
```diff
@@ -92,4 +94,32 @@ contract TestIntegrationMorphoGetters is IntegrationTest {
...
+        uint256 liquidity = market.liquidity();
+        flashBorrower.flashLoanSimple(market.underlying, liquidity);
+
+        uint256 poolSupplyIndexAfter = morpho.updatedIndexes(market.underlying).supply.poolIndex;
+
+        assertGt(poolSupplyIndexAfter, poolSupplyIndexBefore);
+    }
+
+    function testP2PSupplyIndexGrowthInsideBlock(uint256 seed) public {
+        TestMarket storage market = testMarkets[_randomCollateral(seed)];
```
Comment (Tristan22400) [inline test/integration/TestIntegrationMorphoGetters.sol:113]: same here

File: test/integration/TestIntegrationWithdrawCollateral.sol [hunk 3]
```diff
@@ -311,4 +313,39 @@ contract TestIntegrationWithdrawCollateral is IntegrationTest {
         vm.expectRevert(Errors.CollateralIsZero.selector);
         user.withdrawCollateral(market.underlying, amountToWithdraw, onBehalf, receiver);
     }
+
+    function testShouldNotWithdrawMoreCollateralThanSuppliedWhenFlashLoanInflates(uint256 seed) public {
```
Comment (Tristan22400) [inline test/integration/TestIntegrationWithdrawCollateral.sol:317]: change the name

File: test/integration/TestIntegrationWithdrawCollateral.sol [hunk 4]
```diff
@@ -311,4 +313,39 @@ contract TestIntegrationWithdrawCollateral is IntegrationTest {
...
+        TestMarket storage market = testMarkets[_randomCollateral(seed)];
```
Comment (Tristan22400) [inline test/integration/TestIntegrationWithdrawCollateral.sol:318]: same here

**Review by Tristan22400 — CHANGES_REQUESTED** [review]

File: test/integration/TestIntegrationWithdrawCollateral.sol (hunk 4 above)
Comment (Rubilmax) [reply test/integration/TestIntegrationWithdrawCollateral.sol:318]: This test specifically is about collateral assets!

File: test/integration/TestIntegrationWithdrawCollateral.sol (hunk 3 above)
Comment (Rubilmax) [reply test/integration/TestIntegrationWithdrawCollateral.sol:317]: What for?

File: test/integration/TestIntegrationWithdrawCollateral.sol (hunk 4 above)
Comment (Tristan22400) [reply test/integration/TestIntegrationWithdrawCollateral.sol:318]: ok understood but we should do the same for suppliable assets so.

File: test/integration/TestIntegrationWithdrawCollateral.sol (hunk 3 above)
Comment (Tristan22400) [reply test/integration/TestIntegrationWithdrawCollateral.sol:317]: it was about removing collateral word but it is fine after your comment.

**Review by pakim249CAL — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by MathisGD — APPROVED** [review]

**Pull request merged**